spark/spark-scala-parent/src/main/scala/org/apache/zeppelin/spark/BaseSparkScalaInterpreter.scala

```diff
@@ -79,9 +79,49 @@ abstract class BaseSparkScalaInterpreter(val conf: SparkConf,
     System.setProperty("scala.repl.name.line", ("$line" + this.hashCode).replace('-', '0'))
   }
 
-  protected def interpret(code: String, context: InterpreterContext): InterpreterResult
+  def interpret(code: String, context: InterpreterContext): InterpreterResult = {
+
+    def _interpret(code: String): scala.tools.nsc.interpreter.Results.Result = {
+      Console.withOut(interpreterOutput) {
+        interpreterOutput.setInterpreterOutput(context.out)
+        interpreterOutput.ignoreLeadingNewLinesFromScalaReporter()
+        context.out.clear()
+
+        val status = scalaInterpret(code) match {
+          case success@scala.tools.nsc.interpreter.IR.Success =>
+            success
+          case scala.tools.nsc.interpreter.IR.Error =>
+            if (new String(interpreterOutput.getInterpreterOutput.toByteArray)
+              .contains("value toDF is not a member of org.apache.spark.rdd.RDD")) {
+              // prepend "import sqlContext.implicits._" due to
+              // https://issues.scala-lang.org/browse/SI-6649
+              _interpret("import sqlContext.implicits._\n" + code)
```

Comment:
It is fine to re-interpret the code, as the Error happens for compilation stage, so the code won't run twice.